lib/rebaser-core/src/nats.rs

```diff
@@ -0,0 +1,129 @@
+use si_data_nats::{async_nats, jetstream};
+
+const NATS_REBASER_REQUESTS_STREAM_NAME: &str = "REBASER_REQUESTS";
+const NATS_REBASER_REQUESTS_STREAM_SUBJECTS: &[&str] = &["rebaser.requests.>"];
+const NATS_REBASER_TASKS_STREAM_NAME: &str = "REBASER_TASKS";
+const NATS_REBASER_TASKS_STREAM_SUBJECTS: &[&str] = &["rebaser.tasks.>"];
+
+pub const NATS_HEADER_REPLY_INBOX_NAME: &str = "X-Reply-Inbox";
+
+pub async fn rebaser_tasks_jetstream_stream(
+    context: &jetstream::Context,
+) -> Result<async_nats::jetstream::stream::Stream, async_nats::jetstream::context::CreateStreamError>
+{
+    let prefix = context.metadata().subject_prefix();
+
+    let subjects: Vec<_> = NATS_REBASER_TASKS_STREAM_SUBJECTS
+        .iter()
+        .map(|suffix| subject::nats_subject(prefix, suffix).to_string())
+        .collect();
+
+    let stream = context
+        .get_or_create_stream(async_nats::jetstream::stream::Config {
+            name: nats_stream_name(prefix, NATS_REBASER_TASKS_STREAM_NAME),
+            description: Some("Rebaser tasks".to_owned()),
+            retention: async_nats::jetstream::stream::RetentionPolicy::WorkQueue,
+            discard: async_nats::jetstream::stream::DiscardPolicy::New,
+            max_messages_per_subject: 1,
+            discard_new_per_subject: true,
+            allow_direct: true,
```

Comment:
This tracks with what I've seen with how the system behaves live: one "lock" per change set until the rebaser yields it or the change set is no longer in use (or timed out, etc.).